Allow chosing between TLS and STARTTLS

gluap opened the pull request:
This allows to configure starttls: If ``starttls`` is configured, the mail server is contacted via regular smtp and a TLS session is initiated by sending STARTTLS. TLS is enforced in this case to prevent downgrade attacks, the connection is aborted if no TLS can be negotiated.

If ``starttls`` is set to ``false``, TLS without ``starttls`` is used instead (as is usually the case on port 465).

``secure`` setting from nodemailer is intentionally not exposed in configuration because exposing it without deducing it from requiretls would enable plaintext connections which seems undesireable.

opatut commented on api/config.json.example:
```diff
@@ -8,6 +8,7 @@
     "smtp" : {
       "host": "mail.example.com",
       "port": 587,
+      "starttls": true,
```
Could you default to 465 and Standard TLS, as it is preferable over STARTTLS?

gluap replied:
done as requested

opatut commented on api/src/config.js:
```diff
@@ -17,6 +17,7 @@ const configSchema = Joi.object({
       smtp: Joi.object({
         host: Joi.string().required(),
         port: Joi.number().default(587),
+        starttls: Joi.boolean().default(true),
```
Please adjust the defaults here as well.

gluap replied:
Done as requested

opatut reviewed (changes requested):
Good job, and great idea to allow only the two secure(ish) modes via config, no plaintext.

But please can we use TLS as the default prefered way? Most mail servers nowadays do support it.

gluap commented:
@opatut from my side I added all requested changes (and they all made sense, thank you for the review) and I squashed my commits to mage a lean pull request. I am posting again because this is my first time actually using the github review feature and I'm unsure whether I need to do something technical to get the "changes requested" to go away or whether the next step is human intervention on your part.

Please notify me if more intervention from my side is required. I'm not trying to push, I'm just worried that the workflow is waiting for me to do something that I don't know about. So if nothing needs to be done on my part please don't feel pushed but take your time.

opatut reviewed (approved):
Good job, thanks for the contribution!
